prow/spyglass/lenses/html/html.go

```diff
@@ -102,14 +104,66 @@ func (lens Lens) Body(artifacts []api.Artifact, resourceDir string, data string,
 	return buf.String()
 }
 
+// extractDocumentDetails parses the HTML to extract the title and
+// meta description tag, if present.
+func extractDocumentDetails(name string, content []byte) document {
+	doc := document{
+		Filename: name,
+		Title:    name,
+		Content:  string(content),
+	}
+
+	description := ""
+	token := html.NewTokenizer(bytes.NewReader(content))
+	isTitle := false
+	for {
+		switch token.Next() {
+		case html.ErrorToken:
+			doc.Content = injectHeightNotifier(doc.Content, name)
+			// Escape double quotes as we are going to put this into an iframes srcdoc attribute. We can not reference the
+			// src directly because we have to inject the height notifier.
+			// Ref: https://html.spec.whatwg.org/multipage/iframe-embed-object.html#attr-iframe-srcdoc
+			doc.Content = strings.ReplaceAll(doc.Content, `"`, `&quot;`)
+
+			if description != "" {
+				doc.Title = doc.Title + fmt.Sprintf(` <abbr class="icon material-icons" title="%s">info</abbr>`, description)
+			}
+
+			return doc
+		case html.StartTagToken, html.SelfClosingTagToken:
+			tt := token.Token()
+			switch tt.Data {
+			case "title":
+				isTitle = true
+			case "meta":
+				content := ""
+				isDescription := false
+				for _, attr := range tt.Attr {
+					if attr.Key == "name" && attr.Val == "description" {
+						isDescription = true
+					} else if attr.Key == "content" {
+						content = attr.Val
+					}
+				}
+				if isDescription {
+					description = content
+				}
+			}
+		case html.TextToken:
+			tt := token.Token()
+			if isTitle {
```

Comment:
Does this behave correctly with empty tag, like `<title></title>` (do we get the `TextToken` with empty string as data so that we can clear `isTitle`)

Reply:
Good point, it'll skip an empty title now